Command/PopulateCommand.php

```diff
@@ -210,49 +203,44 @@ private function populateIndexType(OutputInterface $output, $index, $type, $rese
-                            $options['batch_size'],
-                            count($event->getObjects()),
-                            sprintf('<error>%s</error>', $event->getException()->getMessage())
-                        );
-                    }
-                );
-
-                $this->dispatcher->addListener(
-                    Events::POST_INSERT_OBJECTS,
-                    function(PostInsertObjectsEvent $event) use ($loggerClosure) {
-                        $loggerClosure(count($event->getObjects()), $event->getPager()->getNbResults());
-                    }
-                );
-            }
-            
-            if ($options['ignore_errors']) {
-                $this->dispatcher->addListener(Events::ON_EXCEPTION, function(OnExceptionEvent $event) {
-                    if ($event->getException() instanceof BulkResponseException) {
-                        $event->setIgnore(true);
-                    }
-                });
-            }
-            
-            $provider = $this->pagerProviderRegistry->getProvider($index, $type);
+        if ($loggerClosure) {
+            $this->dispatcher->addListener(
+                Events::ON_EXCEPTION,
+                function(OnExceptionEvent $event) use ($loggerClosure, $options) {
+                    $loggerClosure(
+                        $options['batch_size'],
+                        count($event->getObjects()),
+                        sprintf('<error>%s</error>', $event->getException()->getMessage())
+                    );
+                }
+            );
+
+            $this->dispatcher->addListener(
+                Events::POST_INSERT_OBJECTS,
+                function(PostInsertObjectsEvent $event) use ($loggerClosure) {
+                    $loggerClosure(count($event->getObjects()), $event->getPager()->getNbResults());
+                }
+            );
+        }
 
-            $pager = $provider->provide($options);
+        if ($options['ignore_errors']) {
+            $this->dispatcher->addListener(Events::ON_EXCEPTION, function(OnExceptionEvent $event) {
+                if ($event->getException() instanceof BulkResponseException) {
+                    $event->setIgnore(true);
+                }
+            });
+        }
 
-            $options['indexName'] = $index;
-            $options['typeName'] = $type;
-            $options['batch_size'] = 100;
-            $options['skip_indexable_check'] = false;
+        $provider = $this->pagerProviderRegistry->getProvider($index, $type);
 
-            $this->pagerPersister->insert($pager, $options);
-        } else {
-            $provider = $this->providerRegistry->getProvider($index, $type);
-            $provider->populate($loggerClosure, $options);
-        }
+        $pager = $provider->provide($options);
+
+//        $options['indexName'] = $index;
+//        $options['typeName'] = $type;
+//        $options['batch_size'] = 100;
+//        $options['skip_indexable_check'] = false;
```

Comment:
commiting commented code?

Reply:
The work is still in progress, this code will be removed very soon.